### PR TITLE
implement users table

### PR DESCRIPTION
**Pull request by nberard**
implement users table

runtime without sql : 15min41s
runtime with sql : 9min49s

### CODE REVIEW

**Comment by vincentlepot** [comment]
Perfs on my box:
w/o sql: 10:44.19
w/ sql: 7:18.46

**Comment by dvdn** [comment]
Requirement :
- module psycopg2

File: users.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import psycopg2
+
+start = time()
+
+(source_root, treatment_day_start, treatment_day_end) = common.get_period_from_input()
+(spark, statLines) = common.load_rdd_data(__file__, source_root, treatment_day_start, treatment_day_end)
+new_users = common.get_rdd_loaded_as_dict(statLines)
+
+new_users = new_users.map(
+    # keep usefull data
```
Comment (dvdn) [inline users.py]: wording > useful

**Comment by nberard** [comment]
updated runtimes : 
2min48s with sql

**Comment by dvdn** [comment]
Last commit (on intel Core i5) gives me 368s instead of 742s :haircut_man:

**Comment by Netmisa** [comment]
_Intel core i5_

Start date: `2017-01-17`
End date: `2017-01-17`
Options: `--conf spark.ui.showConsoleProgress=true --master='local[3]'`
Duration: `679s`

```
[spark-stat-analyzer] [OK] [2017-01-31 08:12:39] [2017-01-17] [CanalTP\StatCompiler\Updater\UsersUpdater] [679]
```

```
real	11m20.420s
user	4m0.647s
sys	0m55.397s
```

**Comment by Netmisa** [comment]
_Intel core i5_

Sql: `With`
Start date: `2017-01-17`
End date: `2017-01-17`
Options: `--conf spark.ui.showConsoleProgress=true --master='local[3]' users_sql.py`
Duration: `173s`

```
[spark-stat-analyzer] [OK] [2017-01-31 09:51:44] [2017-01-17] [CanalTP\StatCompiler\Updater\UsersUpdater] [173]
```

```
real	2m54.870s
user	7m52.696s
sys	0m23.325s
```

File: users.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+statLines = common.load_rdd_data(spark, file_list)
+new_users = common.get_rdd_loaded_as_dict(statLines)
+
+new_users = new_users.map(
+    # keep useful data
+    lambda dict: (dict['user_id'], dict['user_name'], dict['request_date'])
+).map(
+    # format to reduce by key
+    lambda (user_id, user_name, request_date): (user_id, (user_name, request_date))
+)
```
Comment (vincentlepot) [inline users.py]: You can gain some time by adding a cache here

**Pull request merged**